### ISSUE TITLE
Custom middleware ignored for aggregating select options

### ISSUE THREAD

**Issue by florianraith** [laravel-filament/filament#14400]
### Package

filament/filament

### Package Version

v3.2.115

### Laravel Version

v11.25.0

### Livewire Version

v3.5.6

### PHP Version

8.3.9

### Problem description

When aggregating options for a select input with a defined relationship, the middlewares assigned in the panel provider are not used for the request. Instead, the `web` middlewares are used.

The request for opening an edit page. Everything works as expected.
<img width="683" alt="image" src="https://github.com/user-attachments/assets/19828398-6111-48cb-9021-e3a67f4b65e7">

The request for updating the select options. As you can see the `web` middleware is applied here instead of the same middleware as above.
<img width="688" alt="image" src="https://github.com/user-attachments/assets/8523e46a-fc56-4a49-91b4-83084f0f4229">

### Expected behavior

The request for aggregating the select options should use the same middlewares defined in the panel provider as the other filament requests. 

### Steps to reproduce

1. Create two models with a one-to-many (or many-to-many) relationship, which I'll refer to as "parent" and "child" (e.g., posts and tags, videos and comments, etc.).
2. Create resources for each model, where the parent has a select form component with a relationship to the child model.
3. Add middleware to the panel provider that adds a global scope to the child models.
4. Now edit a parent model and try to select a child model. The middleware is not being applied, thus the global scope is ignored.

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/florianraith/filament-issue-demo

**Comment by balintcodes** [comment]
@florianraith Referring to the second part of the [Applying middleware](https://filamentphp.com/docs/3.x/panels/configuration#applying-middleware) section of the documentation:

> By default, middleware will be run when the page is first loaded, but not on subsequent Livewire AJAX requests. If you want to run middleware on every request, you can make it persistent by passing true as the second argument to the middleware() method:

In your panel provider move the custom middleware to a separate `->middleware()` part. Setting `isPersistent` to `true` will have your middleware applied to each AJAX call (when child models are queried to be shown in the option list).

In summary - based on your example repo - your panel provider should look like this:

```
...

->middleware([
    EncryptCookies::class,
    AddQueuedCookiesToResponse::class,
    StartSession::class,
    AuthenticateSession::class,
    ShareErrorsFromSession::class,
    VerifyCsrfToken::class,
    SubstituteBindings::class,
    DisableBladeIconComponents::class,
    DispatchServingFilamentEvent::class,
])
->middleware([
    SetGlobalTenant::class
], isPersistent: true)

...
```
